provider/datagen/src/transform/cldr/datetime/neo.rs

```diff
@@ -0,0 +1,478 @@
+}
+
+fn weekday_convert(
+    data: &ca::Dates,
+    _calendar: &Value,
+    context: Context,
+    length: Length,
+) -> Result<LinearSymbolsV1<'static>, DataError> {
+    let day_symbols = data.days.load(context, length);
+
+    let days = [
+        &*day_symbols.sun,
+        &*day_symbols.mon,
+        &*day_symbols.tue,
+        &*day_symbols.wed,
+        &*day_symbols.thu,
+        &*day_symbols.fri,
+        &*day_symbols.sat,
+    ];
+
+    Ok(LinearSymbolsV1 {
+        symbols: (&days).into(),
+    })
+}
+
+fn dayperiods_convert(
+    data: &ca::Dates,
+    _calendar: &Value,
+    context: Context,
+    length: Length,
+) -> Result<LinearSymbolsV1<'static>, DataError> {
+    let day_periods = data.day_periods.load(context, length);
+
+    let mut periods = vec![&*day_periods.am, &*day_periods.pm];
+
+    if let Some(ref noon) = day_periods.noon {
+        periods.push(noon);
+    } else if day_periods.midnight.is_some() {
+        periods.push(""); // blank entry to separate midnight
+    };
+
+    if let Some(ref midnight) = day_periods.midnight {
+        periods.push(midnight)
+    }
+
+    Ok(LinearSymbolsV1 {
+        symbols: (&periods).into(),
+    })
+}
+
+fn years_convert(
+    data: &ca::Dates,
+    calendar: &Value,
+    context: Context,
+    length: Length,
+) -> Result<YearSymbolsV1<'static>, DataError> {
+    assert_eq!(
+        context,
+        Context::Format,
+        "Eras do not participate in standalone formatting"
+    );
+
+    let eras = match length {
+        Length::Abbr => &data.eras.abbr,
+        Length::Narrow => &data.eras.narrow,
+        Length::Wide => &data.eras.names,
+        Length::Short => unreachable!("Years do not have short symbols!"),
+    };
+
+    // Tostring can be removed when we delete symbols.rs, or we can perhaps refactor it to use Value
+    let calendar_str = calendar.to_string();
+    let map = super::symbols::get_era_code_map(&calendar_str);
+    let mut out_eras: BTreeMap<UnvalidatedTinyAsciiStr<16>, &str> = BTreeMap::new();
+
+    for (cldr, code) in map {
+        if let Some(name) = eras.get(&*cldr) {
+            out_eras.insert(code.into(), &**name);
+        }
+    }
+    // Todo: Cyclic years (#3761)
+    Ok(YearSymbolsV1::Eras(out_eras.into_iter().collect()))
+}
+
+fn months_convert(
+    data: &ca::Dates,
+    calendar: &Value,
+    context: Context,
+    length: Length,
+) -> Result<MonthSymbolsV1<'static>, DataError> {
+    let months = data.months.load(context, length);
+
+    // Tostring can be removed when we delete symbols.rs, or we can perhaps refactor it to use Value
+    let calendar_str = calendar.to_string();
+    let (map, has_leap) = super::symbols::get_month_code_map(&calendar_str);
+
+    if has_leap {
+        let mut symbols: BTreeMap<UnvalidatedTinyAsciiStr<4>, &str> = BTreeMap::new();
+        for (k, v) in months.0.iter() {
+            let code = if k == "7-yeartype-leap" && *calendar == value!("hebrew") {
+                tinystr!(4, "M06L")
```

Comment:
Nit before merge: make sure this generates the right table for Hebrew. We spent a while getting this right with Pedro.